deploy-web/src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -85,29 +93,43 @@ export const KeplrWalletProvider = ({ children }) => {
 
   useEffect(() => {
     if (isWindowLoaded && isSettingsInit) {
-      if (!!window.keplr) {
-        setIsKeplrInstalled(true);
+      if (!!window.keplr || !!window.leap) {
+        if (!!window.keplr) {
+          setIsKeplrInstalled(true);
+        }
+
+        if (!!window.leap) {
+          setIsLeapInstalled(true);
+        }
+
+        if (localStorage.getItem("wallet_autoconnect")) {
+          const storedWallet = localStorage.getItem("wallet_autoconnect");
 
-        window.keplr.defaultOptions = {
-          sign: {
-            preferNoSetMemo: true
+          if (storedWallet === Wallets.KEPLR) {
+            window.wallet = window.keplr;
+          } else if (storedWallet === Wallets.LEAP) {
+            window.wallet = window.leap;
           }
-        };
 
-        if (localStorage.getItem("keplr_autoconnect")) {
+          window.wallet.defaultOptions = {
+            sign: {
+              preferNoSetMemo: true
+            }
+          };
+
           loadWallet();
         } else {
           setIsWalletLoaded(true);
         }
 
-        window.addEventListener("keplr_keystorechange", onKeystoreChange);
+        window.addEventListener("keplr_keystorechange", () => onKeystoreChange(Wallets.KEPLR));
+        window.addEventListener("leap_keystorechange", () => onKeystoreChange(Wallets.LEAP));
 
         return () => {
           isMounted.current = false;
 
-          console.log("useKeplr on unmount");
-
-          window.removeEventListener("keplr_keystorechange", onKeystoreChange);
+          window.removeEventListener("keplr_keystorechange", () => onKeystoreChange(Wallets.KEPLR));
+          window.removeEventListener("leap_keystorechange", () => onKeystoreChange(Wallets.LEAP));
```

Comment:
remove/add listeners must use the same reference to properly work, so we could just set the functions per wallet in respective variables just before adding the listeners.